### PR TITLE
Add additional primitive shapes

### PR DESCRIPTION
**Pull request by misslivirose**
Additional code to support:
- Cylinder 
- 2D Circle
- Cone

Test plan:
- Create a box or sphere entity
- In the entity properties tab, change the Shape property to the new types (Cylinder, circle, cone) and confirm that the geometry updates

Look for: 
- Whether or not the z-index is properly working for the UI. I tested this in desktop mode and tablet mode in HMD, but might have had a caching issue where the drop down was still disappearing behind the User Data JSON editor

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/432/HighFidelity-Beta-PR10624-5579c409e77f4034e13cbb85c6d5083bc9dc97bf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/432/HighFidelity-Beta-PR10624-5579c409e77f4034e13cbb85c6d5083bc9dc97bf.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

The cylinder works but I cannot get any further down the list. **Fail**
image

Please let me know if there is anything more I can do/provide to assist. :)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/443/HighFidelity-Beta-PR10624-6e5cb63d4f8c2daf18fc015608e5f643e4ea917a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/443/HighFidelity-Beta-PR10624-6e5cb63d4f8c2daf18fc015608e5f643e4ea917a.dmg

**Comment by YuppyPlays** [comment]
Confirmed with dev for testing, Testing now.

**Comment by YuppyPlays** [comment]
Testing complete:

I can confirm all works as expected :) See image below for more info. **Pass**

image

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/475/HighFidelity-Beta-PR10624-c3bf915fd4d289584130425a8b6543c345bd99fa.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/475/HighFidelity-Beta-PR10624-c3bf915fd4d289584130425a8b6543c345bd99fa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/488/HighFidelity-Beta-PR10624-34bd2b8a2b93140b4b8f209eb081a6a4ebc67eb3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/488/HighFidelity-Beta-PR10624-34bd2b8a2b93140b4b8f209eb081a6a4ebc67eb3.dmg

File: libraries/render-utils/src/GeometryCache.cpp [hunk 1]
```diff
@@ -256,17 +256,25 @@ void setupSmoothShape(GeometryCache::ShapeData& shapeData, const geometry::Solid
...
 
-    // Top and bottom faces
+    // Top (if not conical) and bottom faces
     std::vector<vec3> shape = polygon<N>();
-    for (const vec3& v : shape) {
-        vertices.push_back(vec3(v.x, 0.5f, v.z));
-        vertices.push_back(vec3(0, 1, 0));
+    if (isConical) {
+        for (uint32_t i = 0; i < N; i++) {
+            vertices.push_back(vec3(0, 0.5f, 0));
```
Comment (sethalves) [inline libraries/render-utils/src/GeometryCache.cpp]: coding standard requires `vec3(0.0f, 0.5f, 0.0f)` here and in various other places in this PR

File: libraries/render-utils/src/GeometryCache.cpp [hunk 2]
```diff
@@ -325,6 +333,39 @@ void extrudePolygon(GeometryCache::ShapeData& shapeData, gpu::BufferPointer& ver
...
+    VertexVector vertices;
+    IndexVector solidIndices, wireIndices;
+
+    std::vector<vec3> shape = polygon<64>();
+    for (const vec3& v : shape) {
+        vertices.push_back(vec3(v.x, 0, v.z));
+        vertices.push_back(vec3(0, 0, 0));
+    }
+
+    for (uint32_t i = 2; i < 64; ++i) {
```
Comment (sethalves) [inline libraries/render-utils/src/GeometryCache.cpp]: 64 here and below should probably be in a named constant

File: libraries/render-utils/src/GeometryCache.cpp (hunk 1 above)
Comment (misslivirose) [inline libraries/render-utils/src/GeometryCache.cpp]: Just to confirm before committing fixes, for consistency, if vertex values are all whole numbers, should those also get changed, or only cases where one value is a float?

Comment (sethalves) [inline libraries/render-utils/src/GeometryCache.cpp]: The rule is this one: [Never rely on implicit type conversion.](https://github.com/highfidelity/hifi/wiki/Coding-Standard#312--never-rely-on-implicit-type-conversion--not-floatvalue--intvalue).  If the values will end up being floats, they should have the .0f on them.

Comment (misslivirose) [inline libraries/render-utils/src/GeometryCache.cpp]: Okay, cool. From what I can tell, the ones that are all ints don't seem to be getting converted anywhere so I'll leave them in their original state and fix the rest.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/544/HighFidelity-Beta-PR10624-1808505d8b588912643ad3856a9c419ad6f9c47c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/544/HighFidelity-Beta-PR10624-1808505d8b588912643ad3856a9c419ad6f9c47c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/548/HighFidelity-Beta-PR10624-8c488d95d1a097eb36359c667c821c376a63eabe.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/548/HighFidelity-Beta-PR10624-8c488d95d1a097eb36359c667c821c376a63eabe.dmg

File: libraries/render-utils/src/GeometryCache.cpp [hunk 3]
```diff
@@ -256,31 +256,39 @@ void setupSmoothShape(GeometryCache::ShapeData& shapeData, const geometry::Solid
...
-    for (const vec3& v : shape) {
-        vertices.push_back(vec3(v.x, 0.5f, v.z));
-        vertices.push_back(vec3(0, 1, 0));
+    if (isConical) {
+        for (uint32_t i = 0; i < N; i++) {
+            vertices.push_back(vec3(0.0f, 0.5f, 0.0f));
+            vertices.push_back(vec3(0.0f, 1.0f, 0.0f));
+        }
+    }
+    else {
```
Comment (sethalves) [inline libraries/render-utils/src/GeometryCache.cpp]: put this else on the previous line with the }

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/553/HighFidelity-Beta-PR10624-7785473ccbe3317adbf9e43516dde49b7bfc4a16.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/553/HighFidelity-Beta-PR10624-7785473ccbe3317adbf9e43516dde49b7bfc4a16.dmg

**Pull request merged**